### DISCUSSION TITLE
Plans on Supporting Remote Control for Airplay2?

### DISCUSSION THREAD

**Discussion by YutongGu** [mikebrady/shairport-sync#1555] [Ideas]
First of all, I want to say that I love the work you are doing. I'm very excited about the new Airplay2 support. I just wanted to see if remote control is something on the roadmap and if so, when we might be able to expect it? Thanks!

**Comment by mikebrady** [comment]
Thanks for the kind words. Remote Control would be great, but we don't yet have any firm plans. Getting an initial release of Shairport Sync out (which should be Real Soon Now) is the main priority right now. The real issue with Remote Control is that there is no documentation -- the protocols have not been made available.

**Comment by danpatterson73** [comment]
hi, 
I am a very loyal Juke customer (their Juke Audio box uses the shairport-sync code). However, not having remote control via phone or Apple Watch is a big limitation. This is really one of the main benefits of Airplay - being able to control different devices that are already playing from say your watch. Anything you can do to make this a priority would make a big difference to the Juke community. 
Thanks
Dan.

**Comment by SZE2** [comment]
Hi Mike! Thanks for making this. 
I was wondering if perhaps the remote control abilities were added or are even able to be added? I too have a Juke amp that uses Shairport Sync to enable airplay 2, and wuld love to be able to remote control them as well.

**Comment by mikebrady** [comment]
The short answer is yes! If you move over to the development version, the same remote control commands, more or less, should now work on AirPlay 2 clients, including transport control and volume. We hope to get MPRIS progress very soon too. It might all be a little rough at the moment.

**Reply by karmaisacat** [reply]
Does this mean a shairport server will appear under the "All Speakers & TVs" menu? (pictured below) 

<img width="1179" height="2556" alt="Screenshot 2026-08-27 at 8 00 23 PM" src="https://github.com/user-attachments/assets/6864fa28-f540-44c7-acc5-014872c3c5c7" />
<img width="1179" height="2556" alt="Screenshot 2026-08-27 at 8 00 37 PM" src="https://github.com/user-attachments/assets/090fe691-3164-4eec-bdf9-66824eb0c081" />

**Comment by mikebrady** [comment]
Well, the answer is a bit complicated.

Generally, only _sources_ of audio show up here, e.g. the TVs, the phone and the Home Pod ("HP").

However, if you do the following slightly complicated thing:
1. Start playing something on a HP and add a Shairport Sync device as an extra speaker ("SPS"), so you have two speakers playing -- the "HP" and the "SPS".
2. Now deselect the "HP", so you're left with "SPS" playing the audio.

While the SPS device is playing (and apparently still connected to the HP), it  will appear in the "All Speakers & TV". Strange but true. 

This works with existing Shairport Sync -- the new features don't affect it.
